pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -980,14 +980,24 @@ func validateVSphere(m *Machine, clusterID string) (bool, []string, utilerrors.A
 
 	errs = append(errs, validateVSphereNetwork(providerSpec.Network, field.NewPath("providerSpec", "network"))...)
 
-	if providerSpec.NumCPUs < minVSphereCPU {
-		warnings = append(warnings, fmt.Sprintf("providerSpec.numCPUs: %d is less than the minimum value (%d): the minimum value will be used instead", providerSpec.NumCPUs, minVSphereCPU))
+	if providerSpec.NumCPUs < minVSphereCPU && providerSpec.NumCPUs != 0 {
+		warnings = append(warnings, fmt.Sprintf("providerSpec.numCPUs: %d is less than the minimum value (%d)", providerSpec.NumCPUs, minVSphereCPU))
 	}
-	if providerSpec.MemoryMiB < minVSphereMemoryMiB {
-		warnings = append(warnings, fmt.Sprintf("providerSpec.memoryMiB: %d is less than the recommended minimum value (%d): nodes may not boot correctly", providerSpec.MemoryMiB, minVSphereMemoryMiB))
+	if providerSpec.MemoryMiB < minVSphereMemoryMiB && providerSpec.MemoryMiB != 0 {
+		warnings = append(warnings, fmt.Sprintf("providerSpec.memoryMiB: %d is less than the recommended minimum value (%d)", providerSpec.MemoryMiB, minVSphereMemoryMiB))
 	}
-	if providerSpec.DiskGiB < minVSphereDiskGiB {
-		warnings = append(warnings, fmt.Sprintf("providerSpec.diskGiB: %d is less than the recommended minimum (%d): nodes may fail to start if disk size is too low", providerSpec.DiskGiB, minVSphereDiskGiB))
+	if providerSpec.DiskGiB < minVSphereDiskGiB && providerSpec.DiskGiB != 0 {
+		warnings = append(warnings, fmt.Sprintf("providerSpec.diskGiB: %d is less than the recommended minimum (%d)", providerSpec.DiskGiB, minVSphereDiskGiB))
+	}
+
+	if providerSpec.NumCPUs == 0 {
+		errs = append(errs, field.Invalid(field.NewPath("providerSpec", "numCPUs"), providerSpec.NumCPUs, "numCPUs must be greater than 0"))
+	}
```

Comment:
IIRC there's something in the controller that caters for this scenario, can we check that before we add this one? 

We want to verify that placing a 0 value in this field will result in a failed machine, have you tried this for each of the three errors you're adding?